pilot/pkg/model/network.go

```diff
@@ -267,7 +269,7 @@ func (mgr *NetworkManager) IsMultiNetworkEnabled() bool {
 	}
 	mgr.mu.RLock()
 	defer mgr.mu.RUnlock()
-	return len(mgr.byNetwork) > 0
+	return mgr.multiNetworkEnabled
```

Comment:
why do we need this change?

Reply:
Please see https://github.com/istio/istio/issues/44155#issue-1645326965
> When resolving addresses, the NetworkManager first [does this](https://github.com/istio/istio/blob/4157e1887071037b369fd82cb4559dc7628bb57b/pilot/pkg/model/network.go#L221), and then tries to [add the gateway info back](https://github.com/istio/istio/blob/4157e1887071037b369fd82cb4559dc7628bb57b/pilot/pkg/model/network.go#L244) once it has resolved the address. However, if the IP address cannot be resolved, the gateway info is not added back. This leaves the NetworkManager in a status where it thinks [multinetwork is disabled](https://github.com/istio/istio/blob/4157e1887071037b369fd82cb4559dc7628bb57b/pilot/pkg/model/network.go#L270), causing stiod to [push all endpoints](https://github.com/istio/istio/blob/4157e1887071037b369fd82cb4559dc7628bb57b/pilot/pkg/xds/ep_filters.go#L39-L42) to the sidecar.

Reply:
What does it send when it failed to resolve hostname with this patch?

Reply:
If the resolve never succeed, then what does it send?

Reply:
Then it only will only send endpoints in the same network, without ew-gateways
https://github.com/istio/istio/blob/1.17.1/pilot/pkg/xds/ep_filters.go#L117